### PR TITLE
Refactor: Build parameterised test class for subclasses of BaseModel

### LINKED ISSUE #159: Build parameterised test class for subclasses of BaseModel
## User stories
As a developer I want to use a parameterised test class for my tests in order to minimize duplicate code.

## Acceptance Criteria
- [x] Create parameterised test class
- [x] write creation test
   - [x] with empty data
   - [x] only with necessary data
   - [x] with complete data
- [x] write serialization test
- [x] write deserialization test

### PR DESCRIPTION
**Pull request by antonykamp**
Fixes #159 

Replace testclasses for Token, Run, SimulationConfiguration with parametrization.
Move and rename test file for basic table tests.

## PR checklist

- [x] Acceptance criteria fulfilled
- [x] Additional features are tested
- [x] Docs (code, wiki & diagrams) updated
- [x] Breaking changes anounced
- [ ] Dev-branch has been merged into local branch to resolve conflicts
- [ ] Tests and linter have passed AFTER local merge
- [ ] Another dev reviewed and approved

### CODE REVIEW

File: tests/test_table_basic.py
Comment (CR1337) [inline tests/test_table_basic.py:1]: Could you please add type hints to the parameters of the test cases? :)

**Review by CR1337 — CHANGES_REQUESTED** [review]
Good job, see my comment.

File: tests/test_table_basic.py
Comment (CR1337) [reply tests/test_table_basic.py:1]: Or don't we use type hints in test cases?

Comment (antonykamp) [reply tests/test_table_basic.py:1]: Good point!

Comment (antonykamp) [reply tests/test_table_basic.py:1]: Thanks @instant-sky for implementing them 💚

Comment (CR1337) [reply tests/test_table_basic.py:1]: Good job. Thx :)

**Review by CR1337 — APPROVED** [review]
I approve

**Comment by antonykamp** [comment]
I think we should wait for #180 in order to use it here 🤝

**Comment by antonykamp** [comment]
Let's merge 🥳

**Pull request merged**